BRANCHING.md

```diff
@@ -0,0 +1,167 @@
+Branching and Releasing
+=======================
+
+`master` is the primary development branch in `mobilecoin.git`.
+
+Changes are created via PRs, reviewed, and squash-merged into `master`.
+
+Sometimes long running feature branches are created. These are, still, eventually
+squash merged into `master`.
+
+Release branches
+----------------
+
+Major releases are created by creating a release branch e.g. `release/v4.0` off
+of master.
+
+Minor releases are created by creating a release branch e.g. `release/v4.1` off of
+the predecessor `release/v4.0`.
+
+Any bugs that are found are fixed in the release branch first.
+
+Release candidates are tagged on release branches.
+
+Propagating changes
+-------------------
+
+Changes which are made in release branches are propagated forward to master,
+by *merging* the release branch into master.
```

Comment:
:question: It seems like the changelog is a manual at release time creation. Will this have any negative affects generating it?  
I wouldn't think so, as long as the history for release N is looked at for the changelog for release N, it will make it to master in the merge and then the master side of the history can be looked at to get the "additional" changes.

Reply:
I think we were trying to get into a pattern where if you merge a notable change, you also make a changelog entry. This would ensure that these changes are merged forward into master and not abandoned in the release branch